hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/analysis/HoodieAnalysis.scala

```diff
@@ -181,6 +183,67 @@ object HoodieAnalysis extends SparkAdapterSupport {
    */
   case class AdaptIngestionTargetLogicalRelations(spark: SparkSession) extends Rule[LogicalPlan] {
 
+    /**
+     * The function updates the HadoopFSRelation so that it uses HoodieFileIndex instead of
+     * HoodieFileIndexTimestampKeyGen. Also the data type for output attributes of the plan are changed
```

Comment:
`HoodieFileIndexTimestampKeyGen` -> i don't see this class anywhere used. Did you mean to use `HoodieReaderFileIndex`?

Reply:
Addressed